server/src/test/java/org/elasticsearch/index/mapper/KeywordFieldMapperTests.java

```diff
@@ -789,157 +788,167 @@ public void testFieldTypeWithSkipDocValues_LogsDbMode() throws IOException {
-
-        final MapperService mapperService = createMapperService(
-            Settings.builder()
-                .put(IndexSettings.MODE.getKey(), IndexMode.LOGSDB.name())
-                .put(IndexSortConfig.INDEX_SORT_FIELD_SETTING.getKey(), "hostname")
-                .build(),
-            mapping(b -> {
-                b.startObject("hostname");
-                b.field("type", "keyword");
-                b.endObject();
-            })
-        );
+        final Settings settings = Settings.builder()
+            .put(IndexSettings.MODE.getKey(), IndexMode.LOGSDB.name())
+            .put(IndexSortConfig.INDEX_SORT_FIELD_SETTING.getKey(), "hostname")
+            .build();
+        final MapperService mapperService = createMapperService(settings, mapping(b -> {
+            b.startObject("hostname");
+            b.field("type", "keyword");
+            b.endObject();
+        }));
 
         final KeywordFieldMapper mapper = (KeywordFieldMapper) mapperService.documentMapper().mappers().getMapper("hostname");
         assertTrue(mapper.fieldType().hasDocValues());
         assertTrue(mapper.fieldType().isIndexed());
-        assertFalse(mapper.fieldType().hasDocValuesSparseIndex());
+        assertFalse(mapper.fieldType().hasDocValuesSkipper());
     }
 
-    public void testFieldTypeDefault_ConfiguredIndexed() throws IOException {
-        assumeTrue("Needs feature flag to be enabled", FieldMapper.DOC_VALUES_SPARSE_INDEX.isEnabled());
+    public void testFieldTypeDefault_ConfiguredIndexedWithSettingOverride() throws IOException {
+        final Settings settings = Settings.builder()
+            .put(IndexSettings.MODE.getKey(), IndexMode.LOGSDB.name())
+            .put(IndexSortConfig.INDEX_SORT_FIELD_SETTING.getKey(), "host.name")
+            .build();
+        final MapperService mapperService = createMapperService(settings, mapping(b -> {
+            b.startObject("host.name");
+            b.field("type", "keyword");
+            b.field("index", true);
+            b.endObject();
+        }));
 
-        final MapperService mapperService = createMapperService(
-            Settings.builder()
-                .put(IndexSettings.MODE.getKey(), IndexMode.LOGSDB.name())
-                .put(IndexSortConfig.INDEX_SORT_FIELD_SETTING.getKey(), "host.name")
-                .build(),
-            mapping(b -> {
-                b.startObject("host.name");
-                b.field("type", "keyword");
-                b.field("index", true);
-                b.endObject();
-            })
-        );
+        final KeywordFieldMapper mapper = (KeywordFieldMapper) mapperService.documentMapper().mappers().getMapper("host.name");
+        assertTrue(mapper.fieldType().hasDocValues());
+        assertFalse(IndexSettings.USE_DOC_VALUES_SKIPPER.get(settings) && mapper.fieldType().isIndexed());
+        assertEquals(IndexSettings.USE_DOC_VALUES_SKIPPER.get(settings), mapper.fieldType().hasDocValuesSkipper());
```

Comment:
```suggestion
        if (FieldMapper.DOC_VALUES_SPARSE_INDEX.isEnabled()) {
           assertTrue(mapper.fieldType().hasDocValuesSkipper());
        } else {
          assertFalse(mapper.fieldType().hasDocValuesSkipper());
        }
```

and same for the other similar assertions?